src/hash/hash.c

```diff
@@ -1904,7 +1962,7 @@ size_t fiftyoneDegreesHashSizeManagerFromFile(
  	if (status == SUCCESS && EXCEPTION_OKAY) {
 		ResourceManagerFree(&manager);
 	}
-	else if (status != SUCCESS && EXCEPTION_OKAY) {
+	else if (exception != NULL && status != SUCCESS && EXCEPTION_OKAY) {
```

Comment:
`exception` can not be both:
- not equal to NULL _(as added above)_
- equal to NULL _(as required by `EXCEPTION_OKAY` macro)_

```cpp
#define FIFTYONE_DEGREES_EXCEPTION_OKAY \
(exception == NULL || exception->status == FIFTYONE_DEGREES_STATUS_NOT_SET)
```
^ https://github.com/51Degrees/common-cxx/blob/71f3e1e4e19f9092d1cd341b7c296e0d8497e455/exceptions.h#L139-L143

Reply:
Code and comment modified. The intention of the method is purely to measure the amount of memory that is allocated during initialization. This is needed for environments that require modules to declare the amount of memory they're going to use during initialization. Therefore checking for the exception or throwing the exception are not needed. See commit https://github.com/51Degrees/device-detection-cxx/pull/87/commits/9cf57e5bd8567665ad4d07f62d9fd41cbb33b6c1